File: 0169-majority-element/0169-majority-element.java
import java.util.HashMap;

class Solution {
    public int majorityElement(int[] nums) {
        HashMap <Integer,Integer> map = new HashMap<>();
        for (int n : nums) {
            if (!map.containsKey(n))
                map.put(n,1);
            else {
                int temp = map.get(n);
                map.put(n, ++temp);
            }
        }
        int count = map.get(nums[0]);
        int res = nums[0];
        for (int n : nums){
            if (map.get(n) > count){
                count = map.get(n);
                res = n;
            }
        }
        return res;
    }
}

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
   public int MajorityElement(int[] nums)
{
    int count = 0;
    int target = 0;

    foreach (var item in nums)
    {
        if (count == 0)
        {
            target = item;
        }
        if (target == item)
        {
            count++;
        }
        else
        {
            count--;
        }
    }
    return target;
}
}